go-controller/pkg/ovn/ovn.go

```diff
@@ -1179,3 +1196,107 @@ func shouldUpdate(node, oldNode *kapi.Node) (bool, error) {
 
 	return true, nil
 }
+
+// checkDBMembers checks if the db-ids of the db instances have changed.
+// In case all the db-ids have departed from the original list, we treat
+// this as a need for the master to re-sync all the dbs. This is particularly
+// relevant for when the dbs get corrupted for some unknown reason.
+// This kill-switch lets us recover and resync the dbs to known state.
+func (oc *Controller) checkDBMembers(dbName string) {
+	var appCtl func(args ...string) (string, string, error)
+	var dbIds []string
+	var numDBIdsChanged int
+	if dbName == "OVN_Northbound" {
+		appCtl = util.RunOVNNBAppCtl
+		dbIds = oc.ovnNBDBIds
+	} else {
+		appCtl = util.RunOVNSBAppCtl
+		dbIds = oc.ovnSBDBIds
+	}
+
+	out, stderr, err := appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", dbName, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`([a-z0-9]{4}) at ((ssl|tcp):\[?[a-z0-9.:]+\]?)`)
+	members := r.FindAllStringSubmatch(out, -1)
+	// Currently the dbIds slice is never "updated".
+	// We might need to be able to differentiate between the case where all ids are changing together
+	// as a result of db corruption on nodes, vs. a single db changing at a time.
+	// A single db changing is not worthy of a complete sync by restarting master. However, because we don't
+	// update the db-ids list if each of the three instances change one at a time (over a period of time),
```

Comment:
I don't understand this part of the comment. Can't we keep a running list/number of total instances changed?

Reply:
The point I am trying to make is we can't differentiate between one db-id changing at a time (members getting added/replaced) vs. all db members getting corrupted and getting booted by the ovndbchecker.

Reply:
OK I think I understand now. Maybe reword it as:
"A single db changing is not worth of a complete sync by restarting master. However, over a period of time we may detect 3 instances have changed, and we are unable to determine if they changed in succession and the dbs were sync'ed correctly OR if all 3 instances changed at the same time, and the database is now wiped. Therefore if we see 3 instances change over a period of time, we always exit"